stdlib/src/collections/string/string.mojo

```diff
@@ -1057,7 +1057,6 @@ struct String(
         """
         self._iadd[False](other.as_bytes())
 
-    @deprecated("Use `str.codepoints()` or `str.codepoint_slices()` instead.")
```

Comment:
It serms, that this change is done  "by the way", and is out of scope of this pr